warehouse/query-core/src/main/java/datawave/query/tld/TLDFieldIndexAggregator.java

```diff
@@ -51,7 +51,7 @@ public Key apply(SortedKeyValueIterator<Key,Value> itr, Document d, AttributeFac
             Attribute<?> attr = af.create(field, value, key, true);
             // in addition to keeping fields that the filter indicates should be kept, also keep fields that the filter applies. This is due to inconsistent
             // behavior between event/tld queries where an index only field index will be kept except when it is a child of a tld
-            attr.setToKeep((fieldsToAggregate == null || fieldsToAggregate.contains(JexlASTHelper.removeGroupingContext(field)))
+            attr.setToKeep((fieldsToAggregate == null || fieldsToAggregate.isEmpty() || fieldsToAggregate.contains(JexlASTHelper.removeGroupingContext(field)))
```

Comment:
This class is used in the TLDQueryIterator and the list of non-event fields is explicitly passed in here.  If it is empty, then we should not be keeping any of them.  I believe this change is incorrect.